sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/ForeachSink.scala

```diff
@@ -77,6 +77,7 @@ class ForeachSink[T : Encoder](writer: ForeachWriter[T]) extends Sink with Seria
           case e: Throwable =>
             isFailed = true
             writer.close(e)
+            throw e
         }
         if (!isFailed) {
```

Comment:
I think there is a better of write this now using `try...catch...finally`. Because the condition here will always be true